torchvision/prototype/models/depth/stereo/raft_stereo.py

```diff
@@ -0,0 +1,710 @@
+        self.conv = nn.Conv2d(base_dim, output_dim, kernel_size=1)
+        if shared_base:
+            self.residual_block = block(base_dim, base_dim, norm_layer=nn.InstanceNorm2d, stride=1)
+            self.conv = nn.Conv2d(base_dim, output_dim, kernel_size=3, padding=1)
+
+    def forward(self, x: Tensor) -> Tensor:
+        x = self.base_encoder(x)
+        x = self.residual_block(x)
+        x = self.conv(x)
+        return x
+
+
+class MultiLevelContextEncoder(nn.Module):
+    def __init__(
+        self,
+        base_encoder: nn.Module,
+        out_with_blocks: List[bool],
+        output_dim: int = 256,
+        block: Callable[..., nn.Module] = ResidualBlock,
+    ):
+        super().__init__()
+        self.num_level = len(out_with_blocks)
+        self.base_encoder = base_encoder
+        self.base_downsampling_ratio = base_encoder.downsampling_ratio
+        base_dim = base_encoder.output_dim
+
+        self.downsample_and_out_layers = nn.ModuleList(
+            [
+                nn.ModuleDict(
+                    {
+                        "downsampler": self._make_downsampler(block, base_dim, base_dim) if i > 0 else nn.Identity(),
+                        "out_hidden_state": self._make_out_layer(
+                            base_dim, output_dim // 2, with_block=out_with_blocks[i], block=block
+                        ),
+                        "out_context": self._make_out_layer(
+                            base_dim, output_dim // 2, with_block=out_with_blocks[i], block=block
+                        ),
+                    }
+                )
+                for i in range(self.num_level)
+            ]
+        )
+
+    def _make_out_layer(self, in_channels, out_channels, with_block=True, block=ResidualBlock):
+        block_layer = nn.Identity()
+        conv_layer = nn.Conv2d(in_channels, out_channels, kernel_size=3, padding=1)
+        if with_block:
+            block_layer = block(in_channels, in_channels, norm_layer=nn.BatchNorm2d, stride=1)
+        return nn.Sequential(block_layer, conv_layer)
+
+    def _make_downsampler(self, block, in_channels, out_channels):
+        block1 = block(in_channels, out_channels, norm_layer=nn.BatchNorm2d, stride=2)
+        block2 = block(out_channels, out_channels, norm_layer=nn.BatchNorm2d, stride=1)
+        return nn.Sequential(block1, block2)
+
+    def forward(self, x: Tensor) -> List[Tensor]:
+        x = self.base_encoder(x)
+        outs = []
+        for layer_dict in self.downsample_and_out_layers:
+            x = layer_dict["downsampler"](x)
+            outs.append(torch.cat([layer_dict["out_hidden_state"](x), layer_dict["out_context"](x)], dim=1))
+        return outs
+
+
+class ConvGRU(raft.ConvGRU):
+    """Convolutional Gru unit."""
+
+    # Modified from raft.ConvGRU to accept pre-convolved contexts,
+    # see: https://github.com/princeton-vl/RAFT-Stereo/blob/main/core/update.py#L23
+    def forward(self, h: Tensor, context: List[Tensor], x: Tensor):
```

Comment:
I understand that the current code will have mypy complained (about not compatible with supertype). We purposely revert this back to expose the error on the CI.

Reply:
Here is the error on the CI: https://app.circleci.com/pipelines/github/pytorch/vision/18352/workflows/de86cd36-ddbc-4ea1-a726-30713617abbf/jobs/1484828